engine/common/engine_servicei.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <string>
+#include <vector>
+#include "json/json.h"
+#include "utils/result.hpp"
+
+// TODO: namh think of the other name
+struct DefaultEngineVariant {
+  std::string engine;
+  std::string version;
+  std::string variant;
+
+  Json::Value ToJson() const {
+    Json::Value root;
+    root["engine"] = engine;
+    root["version"] = version;
+    root["variant"] = variant;
+    return root;
+  }
+};
+
+// TODO: namh think of the other name
+struct EngineVariantResponse {
+  std::string name;
+  std::string version;
+  std::string engine;
+
+  Json::Value ToJson() const {
+    Json::Value root;
+    root["name"] = name;
+    root["version"] = version;
+    root["engine"] = engine;
+    return root;
+  }
+};
+
+class EngineServiceI {
```

Comment:
Hmm, I'm not sure if we need this interface.

Reply:
I would like to follow SOLID principle, the `HardwareService` and `ModelService` class does not need to access all function of `EngineService`